File: Python/02is_prime.py


# n = input number
# Time:  O(square_root(n))
# Space: O(1)

from math import sqrt, floor

def is_prime(n):
  if n == 1:
    return False
  
  for i in range (2, floor(sqrt(n)) + 1):
    if n % i == 0:
      return False
    
  return True

File: Javascript/02isPrime.js

// n = input number
// time:  O(square_root(n))
// space: O(1)

const isPrime = (n) => {
  // todo
  if (n < 2) return false;
  for (let i = 2; i <= Math.sqrt(n); i++) {
    if (n % i === 0) return false;
  }
  return true;
};